enigma-principal/app/src/esgx/epoch_keeper_u.rs

```diff
@@ -34,44 +42,59 @@ pub fn set_worker_params(eid: sgx_enclave_id_t, worker_params: &InputWorkerParam
             &mut retval,
             worker_params_rlp.as_c_ptr() as *const u8,
             worker_params_rlp.len(),
+            &seed_in,
+            &nonce_in,
             &mut rand_out,
             &mut nonce_out,
             &mut sig_out,
         )
     };
     if retval != EnclaveReturn::Success || status != sgx_status_t::SGX_SUCCESS {
-        return Err(EnclaveFailError { err: retval, status }.into());
+        bail!("{:?} encountered in the enclave", retval);
```

Comment:
Why did you change that?
`bail!` should be used as a temporary/generic error. `EnclaveFailError` is the error that fit here, and we nmeed both the `status` and the `retval` and you just throw away the `status`